instance/instance_upgrade.py

```diff
@@ -55,11 +56,12 @@ def get_instances_to_upgrade(self):
         """
         Obtains list of instances to be upgraded
         """
-        return OpenEdXInstance.objects.filter(  # Select instances
-            active_appserver___status="running",  # that are running
+        return OpenEdXInstance.objects.filter(Q(  # Select distinct instances
+            Q(ref_set__openedxappserver_set___is_active=True,  # with active appserver(s)
+              ref_set__openedxappserver_set___status="running"),  # that are running
             openedx_release__contains=self.INITIAL_RELEASE,  # on {INITIAL_INSTANCE}
             use_ephemeral_databases=False,  # and use persistent databases.
-        )
+        )).distinct()
```

Comment:
Did you have a look at the resulting SQL?  I can't tell from looking at this code whether this would match instances that have a running appserver and an active appserver, but the active appserver is not running.  I'm also not sure it matters. :)